### ISSUE TITLE
'TestCheckBackwardNoneConvention' is flaky

### ISSUE THREAD

**Issue by Crissman** [chainer/chainer#8095]
[Travis error](https://travis-ci.org/chainer/chainer/jobs/574304849)

FAIL `tests/chainer_tests/test_gradient_check.py::TestCheckBackwardNoneConvention_param_0_{dtype=None, size=3}::test_multiple_output`

Trace:
```__________________ TestCheckBackwardNoneConvention_param_0_{dtype=None, size=3}.test_multiple_output ___________________
2495
2496self = <chainer.testing._bundle.TestCheckBackwardNoneConvention_param_0_{dtype=None, size=3} testMethod=test_multiple_output>
2497args = (), kwargs = {}
2498e = AssertionError('check_backward failed (eps=0.001 atol=0.0001 rtol=0.001)\ninpu....0001\n  total tolerance: 0.000111020203487\nx: -0.01113426\ny: -0.0110202\n',)
2499s = <StringIO.StringIO instance at 0x1581d65a8>, k = 'size', v = 3
2500
2501    @functools.wraps(base_method)
2502    def new_method(self, *args, **kwargs):
2503        try:
2504            return base_method(self, *args, **kwargs)
2505        except unittest.SkipTest:
2506            raise
2507        except Exception as e:
2508            s = six.StringIO()
2509            s.write('Parameterized test failed.\n\n')
2510            s.write('Base test method: {}.{}\n'.format(
2511                base.__name__, base_method.__name__))
2512            s.write('Test parameters:\n')
2513            for k, v in six.iteritems(param2):
2514                s.write('  {}: {}\n'.format(k, v))
2515>           utils._raise_from(e.__class__, s.getvalue(), e)
2516
2517../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/chainer/testing/parameterized.py:89: 
2518_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2519../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/chainer/utils/__init__.py:104: in _raise_from
2520    six.reraise(exc_type, new_exc, sys.exc_info()[2])
2521../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/chainer/testing/parameterized.py:78: in new_method
2522    return base_method(self, *args, **kwargs)
2523tests/chainer_tests/test_gradient_check.py:690: in test_multiple_output
2524    f, (x1, x2), (g1, g2), dtype=self.dtype, atol=1e-4, rtol=1e-3)
2525../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/chainer/gradient_check.py:900: in check_backward
2526    detect_nondifferentiable, is_immutable_params=False
2527../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/chainer/gradient_check.py:464: in run
2528    self._run()
2529../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/chainer/gradient_check.py:507: in _run
2530    self._compare_gradients(gx_numeric, gx_backward, directions)
2531_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2532
2533self = <chainer.gradient_check._CheckBackward object at 0x14f1eda50>, gx_numeric = array(-0.01113426)
2534gx_backward = -0.011020203487251079
2535directions = [array([ 0.1086916 , -0.5573846 , -0.28129812]), array([-0.24138953,  0.28147831,  0.67888214])]
2536
2537    def _compare_gradients(self, gx_numeric, gx_backward, directions):
2538        atol = self.atol
2539        rtol = self.rtol
2540        # Compare the gradients
2541        try:
2542            testing.assert_allclose(
2543                gx_numeric, gx_backward, atol=atol, rtol=rtol)
2544        except AssertionError as e:
2545            eps = self.eps
2546            xs = self.xs
2547            gys = self.gys
2548            f = six.StringIO()
2549            f.write('check_backward failed (eps={} atol={} rtol={})\n'.format(
2550                eps, atol, rtol))
2551            for i, x in enumerate(xs):
2552                f.write('inputs[{}]:\n'.format(i))
2553                f.write('{}\n'.format(x))
2554            for i, gy in enumerate(gys):
2555                f.write('grad_outputs[{}]:\n'.format(i))
2556                f.write('{}\n'.format(gy))
2557            for i, d in enumerate(directions):
2558                f.write('directions[{}]:\n'.format(i))
2559                f.write('{}\n'.format(d))
2560            f.write('gradients (numeric):  {}\n'.format(gx_numeric))
2561            f.write('gradients (backward): {}\n'.format(gx_backward))
2562            f.write('\n')
2563            f.write('x: numeric gradient, y: backward gradient')
2564            f.write(str(e))
2565>           raise AssertionError(f.getvalue())
2566E           AssertionError: Parameterized test failed.
2567E           
2568E           Base test method: TestCheckBackwardNoneConvention.test_multiple_output
2569E           Test parameters:
2570E             dtype: None
2571E             size: 3
2572E           
2573E           
2574E           (caused by)
2575E           AssertionError: check_backward failed (eps=0.001 atol=0.0001 rtol=0.001)
2576E           inputs[0]:
2577E           [0. 1. 2.]
2578E           inputs[1]:
2579E           [0. 1. 2.]
2580E           grad_outputs[0]:
2581E           [1. 1. 1.]
2582E           grad_outputs[1]:
2583E           [1. 1. 1.]
2584E           directions[0]:
2585E           [ 0.1086916  -0.5573846  -0.28129812]
2586E           directions[1]:
2587E           [-0.24138953  0.28147831  0.67888214]
2588E           gradients (numeric):  -0.0111342596938
2589E           gradients (backward): -0.0110202034873
2590E           
2591E           x: numeric gradient, y: backward gradient
2592E           Not equal to tolerance rtol=0.001, atol=0.0001
2593E           
2594E           Mismatch: 100%
2595E           Max absolute difference: 0.00011406
2596E           Max relative difference: 0.01034974
2597E            x: array(-0.011134)
2598E            y: array(-0.01102)
2599E           
2600E           assert_allclose failed: 
2601E             shape: () ()
2602E             dtype: float64 float64
2603E             i: (0,)
2604E             x[i]: -0.0111342596938
2605E             y[i]: -0.0110202034873
2606E             relative error[i]: 0.0103497368874
2607E             absolute error[i]: 0.000114056206539
2608E             relative tolerance * |y[i]|: 1.10202034873e-05
2609E             absolute tolerance: 0.0001
2610E             total tolerance: 0.000111020203487
2611E           x: -0.01113426
2612E           y: -0.0110202
2613
2614../../../.pyenv/versions/2.7.10/lib/python2.7/site-packages/chainer/gradient_check.py:537: AssertionError```

**Comment by kmaehashi** [comment]
Maybe fixed after fixing https://github.com/chainer/chainer/issues/8025?

**Comment by toslunar** [comment]
The issue seems a duplicate of #7973, which is fixed by #7989.
